docs/markdown/DevGuide_BuildProcessor.md

```diff
@@ -0,0 +1,115 @@
+# Build your own Processor
+This guide will walk you through the creation of a new processor.
+
+## Necessary files and changes
+There are three ways to create a new processor in Inviwo:
+1. **Using the GUI**: Just click the `Tools -> Create Sources -> Add Processor` button, navigate to `<module-dir>/src/processors` and specify your processor name in PascalCase (e.g. `MyFancyProcessor`).
+2. **Using the Inviwo-meta-cli**: Check out the Inviwo-meta docs (TODO: link)
+3. **Manually**:
+    In order to create a new processor in Inviwo, you need to create the source files for the processor itself, register it in the according module and add it to the Cmake files.
+    Inviwo-meta does all of those things for you automatically. The following describes the files that need to be created or modified:
+    1. You need to create the processors source and header files in `<module-dir>/src/processors/`
+    2. The processor needs to be registered in its module. To do so, open the `<module-dir>/src/<module-name>module.cpp` and include your processor's header, then add the following line to the module's constructor:
+    ```
+        registerProcessor<MyNewProcessor>();
+    ```
+    3. Add your new processor's header and source to the module's `CMakeLists.txt` (`<module-dir>/CMakeLists.txt`)
+
+We suggest using the GUI whenever possible.
+
+
+## Processor Subclassing
+If you have created your new processor using the GUI or Inviwo-meta, you can see there is already a class definition in your created files. If you created the files manually, make sure your class is implemented as follows:
```

Comment:
I would move this text, about the newly created processor, up to processor creation